### PR TITLE
Fixed part of #2, simplified spec

### PR DESCRIPTION
**Pull request by stephenmac7**
See [this](http://packassembler.github.io/QuickModDoc/qm_spec.html) if you want to know what it looks like now.

### CODE REVIEW

File: qm_spec.markdown [hunk 1]
```diff
@@ -17,48 +17,30 @@ JSON file
...
-description  | string         | no        | A description of the mod
-license      | string         | no        | The mods license
-urls         | object\*\*\*   | no        | An object of URL types
-updateUrl    | qmurl\*\*      | yes       | This URL should point at the QuickMod file, and is used for updating it
-tags         | stringlist     | no        | A list of tags that apply to the mod
-categories   | stringlist     | no        | A list of categories that apply to the mod. Tags and categories might often be more or less the same.
-authors      | object\*\*\*\* | no        | Authors that have participated in the creation of the mod
-references   | object\*\*\*\*\* | no        | A list of all mods that are in some way referenced in the versions file
-versions     | array          | yes       | A list of versions, see [here](qm_versions_spec.markdown) for documentation.
-mavenRepos   | stringlist     | no        | A list of maven base urls to search for if a downloadType is maven, or for maven dependencies
```
Comment (02JanDal) [inline qm_spec.markdown:37]: Why are you removing this one?

File: qm_spec.markdown [hunk 2]
```diff
@@ -17,48 +17,30 @@ JSON file
...
-### mcf:&lt;topic-id&gt;
-
-Expands to http://www.minecraftforum.net/topic/&lt;topic-id&gt;-
+Name          | Type           | Required? | Description
+------------- | -------------- |:---------:| ------------
+formatVersion | int            | yes       | The version of the QuickMod format. Currently 1.
+updateUrl     | string         | yes       | This URL should point at itself, and is used for updating it
+uid           | string         | yes       | A java package style id of the QuickMod. Has to be globally unique to the QuickMod, and should __never__ change. `<author>.<modid>` is a good UID, but remember it may never change as long as it's still the same mod!
+repo          | string         | yes       | A java package style repository id of the QuickMod. Has to be globally unique to your repository, and __never__ change. Two QuickMods with the same UID and repo are considered exactly equal and will overwrite each other.
+versions      | [version]      | yes       | A list of [versions](qm_versions_spec.html)
```
Comment (02JanDal) [inline qm_spec.markdown]: Haskell notation? Please use something easier to understand.

File: qm_spec.markdown [hunk 3]
```diff
@@ -69,17 +51,16 @@ Note: Most URLs etc in the below example won't work
     "uid": "biomesoplenty.BiomesOPlenty",
     "modId": "BiomesOPlenty",
     "name": "Biomes O' Plenty",
-    "nemName": "BiomesOPlenty",
     "repo": "github.glitchfiend",
     "description": "Biomes O' Plenty is a mod that is designed to give players a better Minecraft world to explore, and more of a reason to explore it in the first place. There are a lot of realistic biomes, some fantasy biomes, and other cool things we've added to the mod.",
     "urls": {
-        "icon": [ "github://Glitchfiend@BiomesOPlenty/icon.png" ],
+        "icon": [ "https://rawgit.com/Glitchfiend/BiomesOPlenty/master/icon.png" ],
```
Comment (02JanDal) [inline qm_spec.markdown:57]: Never heard of rawgit before, very interesting.

File: qm_spec.markdown [hunk 4]
```diff
@@ -106,8 +87,8 @@ Note: Most URLs etc in the below example won't work
             "installType": "forgeMod",
             "urls": [
                 {
-                    "url": "http://adf.ly/391097/http://files.minecraftforge.net/BiomesOPlenty/BiomesOPlenty-universal-1.6.4-1.2.1.434.jar",
-                    "downloadType": "sequential"
+                    "url": "http://files.minecraftforge.net/BiomesOPlenty/BiomesOPlenty-universal-1.6.4-1.2.1.434.jar",
+                    "downloadType": "direct"
```
Comment (02JanDal) [inline qm_spec.markdown]: The example should still use the adfly link with the sequential download type.

File: qm_versions_spec.markdown [hunk 5]
```diff
@@ -40,17 +38,9 @@ Value | Description
 direct | Don't display a browser, just download the URL directly
 parallel | Loaded at the same time as all other`parallel` URLs
 sequential | Loaded one-by-one, needed by adfly or similar URLs
-encoded | See below
```
Comment (02JanDal) [inline qm_versions_spec.markdown:43]: Leave this in for now

File: qm_versions_spec.markdown [hunk 6]
```diff
@@ -40,17 +38,9 @@ Value | Description
...
-maven | The URL is a maven identifier, that's searched for in maven central and the repositories given in the root object.
 
 `parallel`is the default.
 
-## Encoding URLs
```
Comment (02JanDal) [inline qm_versions_spec.markdown:48]: This paragraph should stay too.

File: qm_versions_spec.markdown (hunk 5 above)
Comment (stephenmac7) [inline qm_versions_spec.markdown:43]: Why?

File: qm_spec.markdown (hunk 1 above)
Comment (stephenmac7) [inline qm_spec.markdown:37]: Which one?

File: qm_spec.markdown (hunk 2 above)
Comment (stephenmac7) [inline qm_spec.markdown]: Any suggestions?

File: qm_spec.markdown (hunk 4 above)
Comment (stephenmac7) [inline qm_spec.markdown]: Okay

File: qm_versions_spec.markdown (hunk 5 above)
Comment (02JanDal) [inline qm_versions_spec.markdown:43]: Still don't know if it's actually against the EULA

File: qm_spec.markdown (hunk 2 above)
Comment (02JanDal) [inline qm_spec.markdown]: versionlist, version list, list<version>, versions

File: qm_spec.markdown (hunk 1 above)
Comment (02JanDal) [inline qm_spec.markdown:37]: mavenRepos

Comment (stephenmac7) [inline qm_spec.markdown:37]: If you're not using the mavenRepos, then why have it?

File: qm_versions_spec.markdown (hunk 5 above)
Comment (stephenmac7) [inline qm_versions_spec.markdown:43]: Okay, done.

File: qm_spec.markdown (hunk 3 above)
Comment (stephenmac7) [inline qm_spec.markdown:57]: Its like the raw from github but with the right content-type header. They also offer a CDN.

File: qm_versions_spec.markdown [hunk 7]
```diff
@@ -9,25 +9,23 @@ The QuickMod Version objects can contain the following fields:
...
+\*\* Object with the field 'url' and optionally '[downloadType](#downloadtype)' and 'priority', which helps the client select a mirror
 
-\*\*\* Object with the fields 'url', 'downloadType', 'priority', 'hint' and 'group'. 'url' is the URL (can be encoded, see below), 'downloadType' is an enum, see below for possible values, 'priority' is for selecting of mirror and 'hint' and 'group' are related to encoding, see below.
-
-\*\*\*\* an array of objects, where each object has the fields 'name' and (optionally) 'url'. The 'name' should be in the format `<group>:<artifact>:<version>`, 'url' is optional, if supplied it should be an absolute URL, if not the maven repositories supplied in the root object are searched.
+\*\*\* Object with the fields 'name' and 'url'. The 'name' should be in the format `<group>:<artifact>:<version>`, 'url' should be an absolute URL to the library file.
```
Comment (02JanDal) [inline qm_versions_spec.markdown]: url should be the maven repo root

File: qm_versions_spec.markdown (hunk 5 above)
Comment (peterix) [inline qm_versions_spec.markdown:43]: I don't think you can blame the tool for the actions of people who use it. This has valid uses. Also, no Minecraft EULA applies here.

**Pull request merged**